### ISSUE TITLE
The device is very hot since v4.2

### ISSUE THREAD

**Issue by 0hax** [Blackymas/NSPanel_HA_Blueprint#1620]
### TFT Version

4.2

### ESPHome Version

2023.12.5

### Blueprint Version

4.2

### Panel Model

NSPanel EU Model

### What is the bug?

The device is very hot since 4.2 and temperature sensor are thus wrong.

### Steps to Reproduce

Simply flash 4.2 and check temperature sensor. It's 2 degrees more than the normal temperature.
Also you can touch the device and see it's hot.

### Your panel's YAML

```yaml
substitutions:
  wifi_ssid: ""
  wifi_password: ""
  device_name: "salon"
  nextion_update_url: "http://homeassistant.local:8123/local/nspanel_eu.tft"
  heater_relay: "1" # Select 1 for "Relay 1", 2 for "Relay 2" or "0" to a dummy switch/disabled
  temp_min: "8"
  temp_max: "22"
  min_off_time: "300"
  min_run_time: "300"
  min_idle_time: "30"
packages:
  main: !include nspanel_esphome.yaml
  addon_climate: !include nspanel_esphome_addon_climate_heat.yaml
```

**Comment by edwardtfn** [comment]
Mmmm
I haven noticed this, but maybe you should fall back to v4.1.4 and double check the temperature, to make sure this is related to the version.
There is a branch named v4.1.4 with all the old files.

**Comment by 0hax** [comment]
Thanks for your answer.
I tried but I didn't manage to downgrade the blueprint.
Is that possible?

**Comment by hapklaar** [comment]
The internal temp sensor seems to support it is running a bit hotter than before, about 1-1.5 °C. 
I installed 4.2 beta on the 8th in the afternoon:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/7703320/b2a8c4a4-2e0b-4529-a5ce-4b26f95f2094)

**Comment by fabsfuchs** [comment]
Hello,
you can simply replace your nspanel_blueprint.yaml file under automations (file explorer) with the one from the branch. (https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/v4.1.4/nspanel_blueprint.yaml)
Then reload all YAML configurations. Your entity configuration isnt't affected from this.

I just downgrade from v4.2.1 with this method due to connection problems with the nextion display.

**Comment by kkondyl** [comment]
Hello,

I faced the same issue after updating to 4.2.1.
Temperature of panel was at least 3 degrees higher than normal.

I needed to revert esphome package to 4.1.4 to resolve the issue.
I suspect that root cause is the nspanel_esphome_core.yaml and the wifi setting of "power_save_mode: NONE" in 4.2.1.
In 4.1.4 the power save mode is default which is LIGHT for esp32

**Comment by edwardtfn** [comment]
v1.x, v2.x, v3.x and v4.0 always used `power_save_mode: NONE`, only v4.1 used `LIGHT`, so I don't think this is the issue.

You mentioned the temperature is higher, are you measuring the temperature with an external sensor or are you taking this from its embedded temperature sensor?
We had a change on that sensor settings for the first time on v4.2, in order to support Fahrenheit. I think the difference you are seeing is more related to this change.

**Comment by edwardtfn** [comment]
> The internal temp sensor seems to support it is running a bit hotter than before, about 1-1.5 °C. I installed 4.2 beta on the 8th in the afternoon: ![image](https://private-user-images.githubusercontent.com/7703320/296449228-b2a8c4a4-2e0b-4529-a5ce-4b26f95f2094.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzOTUwMjksIm5iZiI6MTcwNTM5NDcyOSwicGF0aCI6Ii83NzAzMzIwLzI5NjQ0OTIyOC1iMmE4YzRhNC0yZTBiLTQ1MjktYTVjZS00YjI2Zjk1ZjIwOTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMTZUMDg0NTI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzNiNDlmZWNjNDhjZDc1ZjI1NDBlYjUyZjQzYTFiNTYzYWQwMzA4Zjk5Yjc3OGIzZTRjOTIzOTA0NjMyZGU4NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.3lU0GxnX0UH9Gq9MdMVYe3lyf9dLo3a9n_bp1hk1JhM)

@hapklaar, do you have a similar trend from your "Temperature correction" sensor?

**Comment by 0hax** [comment]
For me, with v4.2, when touching the screen I definitively saw it was hotter than usual.
After downgrading, I don't have this issue any more.
It's like the esp32 cpu is 100% all time Maybe a bug with newer esphome or with esp-idf framework?

**Comment by kkondyl** [comment]
> v1.x, v2.x, v3.x and v4.0 always used `power_save_mode: NONE`, only v4.1 used `LIGHT`, so I don't think this is the issue.
> 
> You mentioned the temperature is higher, are you measuring the temperature with an external sensor or are you taking this from its embedded temperature sensor? We had a change on that sensor settings for the first time on v4.2, in order to support Fahrenheit. I think the difference you are seeing is more related to this change. Also, are you sure your temperature correction sensor haven't lost it's value with the update? Do you have a trend from that sensor from before and after the upgrade?

I started using nspanel since 4.1 only...
Ambient temperature is around 24.5 C from external thermometer which was inline with nspanel temperature.
After update to 4.2.1 the nspanel temperature had a peak to 29 C and stabilized to around 27.5 C

After downgrade to 4.1.4 the temperature is dropping back to normal:

![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/67463003/1831b7fd-9628-43e0-aa1a-4f3e03ea93f5)

**Comment by MichaelHeimann** [comment]
I do not have any temp correction set since my nspanel with 4.1 and the aquara were close enough.
with 4.2 it changed about 2°K.

![Bildschirmfoto vom 2024-01-16 09-59-33](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/3189562/954fa972-c4e2-4cc5-8101-7138ffd997e4)

**Comment by edwardtfn** [comment]
> I do not have any temp correction set since my nspanel with 4.1

This was available since very early versions, so you should have that with v4.1:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/036e36d5-8587-4d88-81e0-5ad3d4c8758b)

**Comment by MichaelHeimann** [comment]
> > I do not have any temp correction set since my nspanel with 4.1
> 
> This was available since very early versions, so you should have that with v4.1: 

Maybe I was unclear. I have the option, I just didn't need to use it. So it's set to 0.
This was meant as an answer to your question "Also, are you sure your temperature correction sensor haven't lost it's value with the update". Since it was 0 and still is 0 the correction is not the reason for the difference in measurement

**Comment by df9517** [comment]
I had the same problem with 4.2.2. Me device got 5 degrees hotter. 

I think the problem is that the panel wont go into sleeping mode. it stays with screensaver on but dont go to sleep.

**Comment by edwardtfn** [comment]
> I had the same problem with 4.2.2. Me device got 5 degrees warmer after update from 4.1.4.

Are you measuring the device's temperature or are you saying this based on it's embedded temperature sensor? I'm unsure if it could be a thing on that sensor, as that changed with v4.2.
Anyways, looks like the v4.2.2 isn't helping. I've reduced the frequency of watchdog in a try to reduce load, but looks like that didn't made any difference.

> I think the problem is that the panel wont go into sleeping mode. The display goes blank/black but is always on. It dont go to sleep.

That is a good point. On v4.1.4 it don't really goes to sleep also, but brightness was set to 0, while on v4.2 there is a selector for this.
@0hax, @hapklaar, @MichaelHeimann and @df9517, could you please double check how is the entity `number.<<panel name>>_display_brightness_sleep`?

**Comment by df9517** [comment]
I could clearly see that the backlight was on, even the screen was "off" on V4.2.2.

On 4.1.4 it's completely dark.

I'm on 4.1.4 now.

So you think I should change brightness to 0 on 4.2.2 to get same result?

**Comment by MichaelHeimann** [comment]
The panels I have report 2 (more like 2.5) degrees warmer from the internal sensor.
My panels don't go to sleep (timeout sleep = 0,0). 
I do have dimdown brightness set to 5 and display brightness set to 80 so I think most of the time the panel is at 5% brightness.

**Comment by edwardtfn** [comment]
> I do have dimdown brightness set to 5 and display brightness set to 80 so I think most of the time the panel is at 5% brightness.

What about Brightness **Sleep**?

**Comment by MichaelHeimann** [comment]
> > I do have dimdown brightness set to 5 and display brightness set to 80 so I think most of the time the panel is at 5% brightness.
> 
> What about Brightness **Sleep**?

I think I never changed it, just looked and it's at 0.

**Comment by edwardtfn** [comment]
> So you think I should change brightness sleep to 0 on 4.2.2 to get same result? Makes sense, I will try.

Exactly, however 0 should be the default value, so if you haven't touched that, it was probably 0 already.

**Comment by hapklaar** [comment]
>could you please double check how is the entity `number.<<panel name>>_display_brightness_sleep`?

For me this is at 0, I never touched this.

**Comment by rudgej** [comment]
The device is warmer. I have two under room stats ( hindsight is a wonderful thing ) and the temp now over reads so the room is cold !

is there a consensus as to how to bring the temp back down ?

**Comment by edwardtfn** [comment]
> is there a consensus as to how to bring the temp back down ?

I'm working on a fix for this, but if you have to solve it quickly, you can either revert your system back to v4.1.4 or try to temporarily adjust the temperature offset using the "Temperature correction" control at your device's page.

**Comment by Matthijs33** [comment]
Same here, just updated 2 devices to latest 4.2 and got also a rising temperature minute by minute on both.
Reverting back to 4.1.4 tomorrow

**Comment by andythomas** [comment]
TL;DR: If you like 4.1.4 behavior, but want to update to 4.2.2, add the code posted below at the end of your yaml file.

I think, that I found the cause of the temperature increase. 

![422temp](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/995840/2a0780d9-c204-4965-855a-a72b684d1af2)

At the first arrow, I updated from 4.1.4 to 4.2.2 and observed the described temperature increase as well. At the second arrow, I added 
```
wifi:
  power_save_mode: light
```
and flashed it again. Whether the old or the new temperature is 'more correct' is debatable, my offset was 2°C@4.1.4 and 1.3°C@4.2.2 compared to the thermometer I calibrated to the best of my knowledge. So, on the one hand, it needs less offset. On the other hand, it definitely warms the proximity, if the second thermometer is placed directly on top of it, it picks up a temperature increase as well.

**Comment by edwardtfn** [comment]
Just an update here...
I still investigating this and my focus now is on the following:

### Screen saver
I'm investigating this due to the comment https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/1620#issuecomment-1894247215 from @df9517.
The screen saver was changed from v4.1 to v4.2 to enable the display of time on the screen saver page. I can see on all my 3 panels a gap on the temperature around the day this was implemented on `dev`, however my gap is more around 0.5C.
I'm rebuilding the code to make sure nothing is shown on the display when sleeping, unless "Display time on screen saver" is enabled, and added a warning about the possibility of changing temperature when enabling that.

### Wi-Fi power save mode
Based on https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/1620#issuecomment-1893290988 from @kkondyl and the measurements shared by @andythomas on https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/1620#issuecomment-1894528411, this looks to be related to at least part of the temperature change.
The power save mode was `NONE` in all versions prior to v4.1, and we changed that to `LIGHT` at v4.1 to better accommodate Bluetooth components, however that was reverted on v4.2 in a tentative to help with #1478. That didn't solved the issue, but as it was previously tested, I've decided to revert to the mode prior to v4.1.
Based on your findings, I will set this back to `LIGHT`,

## Next steps
I will push a change to `dev` in a few minutes, including changes to the Blueprint, ESPHome and TFT files. If someone could run some tests and share the results I will appreciate a lot, so tomorrow I can use the feedback to either restart my research or to wrap up the patch release.

**Comment by andythomas** [comment]
I tested it the following way:

1) changed the blueprint to `dev`
2) deleted my own modification in the panel's yaml
3) compiled and OTA flashed branch `dev`
4) enabled the `TFT display branch`and uploaded `dev` 

I got the following behavior:
![422Comp3](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/995840/6dbfb727-933b-4970-b164-9187a2fb2225)


Please note that the initial, small peak directly after the update is expected and was [reported](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/1527#issuecomment-1876849999) (by me) also for 4.1.4. The compile, TFT upload and so on supposedly warms the panel a little bit. So if you test it, please give it 15min and do not touch the panel if you want to see, whether the temperature recovers to its original value.  I would say **it is fixed** and wanted to give quick feedback, but I will keep an eye on the temperature development. Please note, that the panel is in my bathroom and me taking a shower and opening the window afterwards leads to some changes. 

However, please note the blue boxes. It did not crunch the numbers, but even eyeballing it shows a reduction of the fluctuations _without_ an increase in the response time due to, e.g., filters. It looks very good!

disclaimer: I am remotely trying the changes and can look at logs, but would not even see obvious bugs with the panel display and interactions.

**Comment by edwardtfn** [comment]
Thanks a lot for your detailed report, @andythomas.

I will wait a little bit more for more info (from you or from others) before pushing a release.
Please let me know if your system keeps stable or not.

**Comment by edwardtfn** [comment]
By the way, I could also see a temperature change on my panels after the update:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/2e778d5a-7ab2-4e37-803e-1e0bb1cd2bf1)

The peak at the end on my testing panel was when I was playing with TFT uploads.

**Comment by andythomas** [comment]
Long(er) term stability is also good. 
![422Comp4](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/995840/82913004-58af-4d7b-a70b-ae57a4d29df3)
Please note the shower/window open parts marked as excluded in light blue. Now, the sun is warming up the apartment and you will see the onset of that, just as yesterday with 4.1.4. @edwardtfn Please wait for feedback from a person who is looking at and interacting with the panel. Also, please note, that I use the optional, embedded climate (only heater) functionality.

**Comment by andythomas** [comment]
And a related question, is it obvious to you (@edwardtfn ), how you decreased the noise floor? This is a welcomed improvement already in 4.2.2. (I am and have always been on `esp-idf`.)

**Comment by edwardtfn** [comment]
> how you decreased the noise floor? This is a welcomed improvement already in 4.2.2. (I am and have always been on esp-idf.

I've done some changes to that sensor to handle Fahrenheit in a better way, but nothing that will intentionally affect the noise level. 👀

**Comment by edwardtfn** [comment]
v4.2.3 was released.
Please let me know if that was fixed, otherwise we can reopen this.

**Comment by df9517** [comment]
Thank you. 4.2.3 is good!
